### ISSUE TITLE
setPageContextPrefetchCache() called on SSG page, with isBrilloutDocpress() false

### ISSUE THREAD

**Issue by lfos** [vikejs/vike#1920]
### Description

~~The following error is sometimes being triggered when using client-side navigation.~~ Edit: Correction - the error actually seems to occur when hovering over internal links (and before clicking).

```
Uncaught (in promise) Error: [vike@0.4.198][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually under 24 hours).
    at createErrorWithCleanStackTrace (createErrorWithCleanStackTrace.js:4:17)
    at assert (assert.js:54:27)
    at setPageContextPrefetchCache (prefetch.js:73:5)
    at prefetchPageContextFromServerHooks (prefetch.js:61:5)
    at async prefetch.js:212:21
    at async Promise.all (index 1)
    at async prefetchOnEvent (prefetch.js:199:5)
```

The assert() is the following from setPageContextPrefetchCache():

```javascript
assert(isBrilloutDocpress()); // Ensure this API isn't used by anyone else
```

I also have random long delays in navigation, likely somewhere between page transition start and the onBeforeRender() hook, which I am currently trying to troubleshoot -- they may or may not be related.

Please let me know if you need more details to investigate/reproduce.

**Comment by lfos** [comment]
FWIW, this is the call stack at the assert():

![20241011_21h57m09s_grim](https://github.com/user-attachments/assets/d57a4d92-b39c-40dc-b4ac-07754ca8ed19)

**Comment by brillout** [comment]
Thank you, I see what's going on. Let me fix this.

**Comment by brillout** [comment]
Fix pre-released as `0.4.198-commit-77f6a27`. Let me know if you still run into any issues.
